src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -110,14 +114,76 @@ private IntPtr AllocException(Exception ex)
         [DllImport(JitSupportLibrary)]
         private extern static char* GetExceptionMessage(IntPtr obj);
 
-        public static void Startup()
+
+        public static void Startup(TargetOS targetOS, TargetArchitecture targetArchitecture, string jitPathOverride)
+        {
+            Debug.Assert(s_jitLibrary == IntPtr.Zero);
+
+            string jitLibraryName;
+            if (!string.IsNullOrEmpty(jitPathOverride))
+            {
+                jitLibraryName = jitPathOverride;
+            }
+            else
+            {
+                string crossgenPath = Path.GetDirectoryName(typeof(CorInfoImpl).Assembly.Location);
+                string jitTargetPath = Path.Combine(crossgenPath, GetTargetSpec(targetOS, targetArchitecture));
+                jitLibraryName = Path.Combine(jitTargetPath, GetLibraryPrefix() + JitLibrary + GetLibraryExtension());
+            }
+
+            s_jitLibrary = NativeLibrary.Load(jitLibraryName);
+            s_jitStartup = Marshal.GetDelegateForFunctionPointer<JitStartupDelegate>(NativeLibrary.GetExport(s_jitLibrary, "jitStartup"));
+            s_getJit = Marshal.GetDelegateForFunctionPointer<GetJitDelegate>(NativeLibrary.GetExport(s_jitLibrary, "getJit"));
+
+            s_jitStartup(GetJitHost(JitConfigProvider.Instance.UnmanagedInstance));
+        }
+
+        private static string GetLibraryPrefix()
+        {
+            return RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "" : "lib";
+        }
+
+        private static string GetLibraryExtension()
         {
-            jitStartup(GetJitHost(JitConfigProvider.Instance.UnmanagedInstance));
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return ".dll";
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                return ".so";
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                return ".dylib";
+            }
+            else
+            {
+                throw new NotImplementedException("GetLibraryExtension");
+            }
+        }
+
+        private static string GetTargetSpec(TargetOS targetOS, TargetArchitecture targetArchitecture)
+        {
+            string targetOSComponent = targetOS switch
+            {
+                TargetOS.Windows => "win",
+                TargetOS.Linux => "linux",
```

Comment:
The JIT .dll is same for `linux`, `osx` and all other Unix systems. Should we have fold linux, osx cases (and all other Unix cases) together into a single name?

Reply:
Good point, will rename to "unix".

Reply:
Fixed in 21st commit.